src/controller/wishcard.ts

```diff
@@ -116,9 +116,13 @@ export default class WishCardController extends BaseController {
 			const wishcards = await this.wishCardRepository.getAll();
 
 			const data = [] as unknown as WishCard & { age: number }[];
-
+			let birthday: moment.Moment;
 			for (let i = 0; i < wishcards.length; i++) {
-				const birthday = moment(new Date(wishcards[i].childBirthday));
+				if (wishcards[i].childBirthYear) {
+					birthday = moment(new Date(wishcards[i].childBirthYear));
+				} else {
+					birthday = moment(new Date(wishcards[i].childBirthday));
+				}
```

Comment:
These changes need to be applied in the handleGetSingle function since that one is called on detail page render

Reply:
@Enubia lol I literally wrote that above 😅